cdap-ai-ext-vertexai/src/test/java/io/cdap/cdap/ai/ext/vertexai/VertexAIConfigurationTest.java

```diff
@@ -0,0 +1,153 @@
+package io.cdap.cdap.ai.ext.vertexai;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Unit tests for the {@link VertexAIConfiguration} class.
+ *
+ * This class contains tests to verify the behavior of the {@link VertexAIConfiguration} class
+ * when initialized with various sets of properties.
+ */
+public class VertexAIConfigurationTest {
+
+  private static final String TEST_PROJECT_ID = "test-project-id";
+
+  private static final String TEST_LOCATION = "test-location";
+
+  private static final String TEST_MODEL_NAME = "test-model";
+
+  private static final String SUMMARY_PROMPT = "Summary prompt";
+
+
+  /**
+   * Tests the {@link VertexAIConfiguration} class when all required properties are provided.
+   *
+   * Verifies that the configuration object correctly initializes and retrieves all properties:
+   * project ID, location, model name, and prompt.
+   */
+  @Test
+  public void testVertexAIConfigurationWithAllProperties() {
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put(VertexAIConfiguration.PROJECT_ID, TEST_PROJECT_ID);
+    properties.put(VertexAIConfiguration.LOCATION_NAME, TEST_LOCATION);
+    properties.put(VertexAIConfiguration.MODEL_NAME, TEST_MODEL_NAME);
+    properties.put(VertexAIConfiguration.Prompts.PIPELINE_MARKDOWN_SUMMARY, SUMMARY_PROMPT);
+
+    VertexAIConfiguration config = new VertexAIConfiguration(properties);
+
+    Assert.assertEquals(TEST_PROJECT_ID, config.getProjectId());
+    Assert.assertEquals(TEST_LOCATION, config.getLocation());
+    Assert.assertEquals(TEST_MODEL_NAME, config.getModelName());
+    Assert.assertEquals(SUMMARY_PROMPT, config.getPrompt().getPipelineMarkdownSummary());
+  }
+
+  /**
+   * Tests the {@link VertexAIConfiguration} class when the project ID property is missing.
+   *
+   * Verifies that the configuration object correctly handles the absence of the project ID
+   * while initializing other properties: location, model name, and prompt.
+   */
+  @Test
+  public void testVertexAIConfigurationWithMissingProjectId() {
```

Comment:
The VertexAIConfiguration constructor should throw exception if the required fields like: `project.id`, `location`, `model` are missing in the map.

Use `Preconditions.checkArgument` to assert that the required keys are present in the map and specify proper error message to throw `IllegalArgumentException`.

Fix this for below tests.

Reply:
Done

Reply:
Done